src/main/java/io/nats/client/support/Validator.java

```diff
@@ -46,23 +47,31 @@ public static String validateJetStreamPrefix(String s) {
     }
 
     interface Check {
-        String check(String s, String label);
+        String check();
     }
 
-    public static String _validate(String s, String label, boolean required, Check check) {
-        if (required) {
-            if (nullOrEmpty(s)) {
-                throw new IllegalArgumentException(label + " cannot be null or empty [" + s + "]");
+    public static String _validate(String s, boolean required, String label, Check check) {
```

Comment:
not necessarily KV related. Just a better implementation than what I originally did.